boltons/mathutils.py

```diff
@@ -0,0 +1,44 @@
+"""This module provides useful math functions on top of Python's built-in :mod:`math` module.
+"""
+
+
+def ceil_from_list(lst, x, allow_equal=True):
+    """
+    Return the ceiling of x, the smallest integer or float from lst that is greater than [or equal to] x.
+
+    Args:
+        lst (list): List of arbitrary numbers (ints or floats).
+        x (int or float): Number to be tested.
+        allow_equal (bool): Defaults to ``True``. Allows equality to be ignored if set to ``False``.
+
+    >>> VALID_CABLE_CSA = [1.5, 2.5, 4, 6, 10, 25, 35, 50]
+    >>> ceil_from_list(VALID_CABLE_CSA, 3.5)
+    4
+    >>> ceil_from_list(VALID_CABLE_CSA, 4)
+    4
+    >>> ceil_from_list(VALID_CABLE_CSA, 4, allow_equal=False)
+    6
+    """
+
+    return min(filter(lambda y: y >= x if allow_equal else y > x, lst))
```

Comment:
btw, using `min()` is far of being the optimal implementation.  you should check this answer http://stackoverflow.com/a/12141511/811740  .